azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java

```diff
@@ -222,6 +246,7 @@
             WithAgentPoolVirtualMachineCount<ParentT>,
             WithMaxPodsCount<ParentT>,
             WithVirtualNetwork<ParentT>,
+            WithMode<ParentT>,
```

Comment:
Is `agentPoolMode` mandatory? If so should it be in a separate stage?

Reply:
It is hard to say mandatory or not. The service requires one pool to be system and others could be default.

Do we prefer to use withSystemMode/withDefaultMode as a mandatory definition?

Reply:
Since the enum is `ExpandableStringEnum` so service could add new one anytime. In this case maybe just an enum would be good enough.

Reply:
After discussing, I think adding it in the optional stage is enough. The service response is enough for customer.